homeassistant/components/modbus/modbus.py

```diff
@@ -249,17 +253,26 @@ async def async_setup(self):
         for entry in self._pb_call.values():
             entry[ENTRY_FUNC] = getattr(self._client, entry[ENTRY_NAME])
 
+        async_call_later(self.hass, 1, self.async_connect_task)
+        return True
+
+    @callback
```

Comment:
```suggestion
```

Reply:
I had a problem that asyc_call_later did not call the function when I removed @callback but I will try and test again this weekend.

Reply:
A coroutine function shouldn't be decorated with `@callback`.

https://developers.home-assistant.io/docs/asyncio_categorizing_functions

Reply:
Found my problem, have removed @callback.